src/sinks/elasticsearch/common.rs

```diff
@@ -140,6 +141,24 @@ impl ElasticsearchCommon {
         })
     }
 
+    /// Parses endpoints.
+    /// Not empty.
+    pub async fn parse_endpoints(config: &ElasticsearchConfig) -> crate::Result<Vec<Self>> {
+        Ok(match config.endpoint {
+            OneOrMany::One(ref endpoint) => vec![Self::parse_config(config, endpoint).await?],
+            OneOrMany::Many(ref endpoints) if endpoints.is_empty() => {
+                return Err(ParseError::EndpointRequired.into())
+            }
+            OneOrMany::Many(ref endpoints) => {
+                let mut commons = Vec::new();
+                for endpoint in endpoints {
+                    commons.push(Self::parse_config(config, endpoint).await?);
+                }
+                commons
```

Comment:
Is it not possible for this case to have specified an empty vector for the endpoints , i.e do we need to check that here and return `EndpointRequired` if the `endpoints` is empty ?

Reply:
That is checked, a few lines above
```
OneOrMany::Many(ref endpoints) if endpoints.is_empty() => {
    return Err(ParseError::EndpointRequired.into())
}
```